cmd/manager/main.go

```diff
@@ -115,16 +120,41 @@ func main() {
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts), zap.StacktraceLevel(zapcore.DPanicLevel)))
 	setupLog.Info("starting up the controller", "version info", version.String())
 
+	var globalPullSecretKey *k8stypes.NamespacedName
+	if globalPullSecret != "" {
+		secretParts := strings.Split(globalPullSecret, "/")
+		if len(secretParts) != 2 {
+			setupLog.Error(fmt.Errorf("incorrect number of components"), "value of global-pull-secret should be of the format <namespace>/<name>")
+			os.Exit(1)
+		}
+		globalPullSecretKey = &k8stypes.NamespacedName{Name: secretParts[1], Namespace: secretParts[0]}
+	}
+
 	if systemNamespace == "" {
 		systemNamespace = podNamespace()
 	}
 
-	dependentRequirement, err := k8slabels.NewRequirement(labels.OwnerKindKey, selection.In, []string{ocv1alpha1.ClusterExtensionKind})
-	if err != nil {
-		setupLog.Error(err, "unable to create dependent label selector for cache")
-		os.Exit(1)
+	cacheOptions := crcache.Options{
+		ByObject: map[client.Object]crcache.ByObject{
+			&ocv1alpha1.ClusterExtension{}: {Label: k8slabels.Everything()},
+			&catalogd.ClusterCatalog{}:     {Label: k8slabels.Everything()},
+		},
+		DefaultNamespaces: map[string]crcache.Config{
+			systemNamespace: {LabelSelector: k8slabels.Everything()},
+		},
```

Comment:
`main` also has `DefaultLabelSelector: k8slabels.Nothing(),` in the cache options. IIRC, that was an important detail.

Reply:
@ankitathomas looks like this needs to fixed?

Reply:
I've already pushed the fix on [line 145](https://github.com/openshift/operator-framework-operator-controller/pull/201/commits/8c0e2123a53ec8090d2c9e3fe1b28d242ff0c4c2#diff-435eecb1d2af40ee96747f88ab71eb2758da989c92f76dcb1f714fc1b300c633R145)

Besides the DefaultLabelSelector @joelanford pointed out, this is exactly the same as the [cacheOptions](https://github.com/openshift/operator-framework-operator-controller/pull/201/commits/8c0e2123a53ec8090d2c9e3fe1b28d242ff0c4c2#diff-435eecb1d2af40ee96747f88ab71eb2758da989c92f76dcb1f714fc1b300c633L136-L145) in use prior to this PR